src/Cli/ConfigGenerator.cs

```diff
@@ -681,6 +686,99 @@ private static bool TryUpdateDepthLimit(
             return true;
         }
 
+        /// <summary>
+        /// Attempts to update the Config parameters in the GraphQL runtime settings based on the provided value.
+        /// Performs the update on the runtimeConfig which is passed as reference
+        /// Returns true if the update has been performed, else false
+        /// </summary>
+        /// <param name="options">Options including the graphql runtime parameters.</param>
+        /// <param name="runtimeConfig">Current config, updated if method succeeds.</param>
+        /// <returns>True if the update was successful, false otherwise.</returns>
+        private static bool TryConfigureRuntimeOptions(
+            ConfigureOptions options,
+            [NotNullWhen(true)] ref RuntimeConfig runtimeConfig)
+        {
+            // GraphQL: Enabled, Path, Allow-Introspection and Multiple-Mutations.Create.Enabled
+            GraphQLRuntimeOptions? updatedGraphQLOptions = runtimeConfig?.Runtime?.GraphQL;
+            if (options.RuntimeGraphQLEnabled != null ||
+                options.RuntimeGraphQLPath != null ||
+                options.RuntimeGraphQLAllowIntrospection != null ||
+                options.RuntimeGraphQLMultipleMutationsCreateEnabled != null)
+            {
+                bool status = UpdateConfigureGraphQLValues(options, ref updatedGraphQLOptions);
+                if (status)
+                {
+                    runtimeConfig = runtimeConfig! with { Runtime = runtimeConfig.Runtime! with { GraphQL = updatedGraphQLOptions } };
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            return runtimeConfig != null;
```

Comment:
even if none of options.RuntimeGraphQLEnabled etc properties are non-null i.e. the if in line 703 returns false, this function will end up saying that it updated the config when the original passed in runtimeConfig != null. This is not what the expected return value should be in such a scenario. Expected return should be false since it is not going to update

Reply:
From what i understood, false is returned when the update is unsuccessful. However, if no update then we should return true? Is this understanding correct? 
With this, the underlying function returns `status`
`bool status = TryUpdateConfigureGraphQLValues(options, ref updatedGraphQLOptions);`
So if there is an error in this the status would be false and false is returned by `TryConfigureRuntimeOptions`, else it would true in all cases where runtime Config is not null.

Reply:
that does make sense when I think in terms of the consumer TryConfigureRuntimeOptions.